### PR TITLE
[wip] Sleepers put occupants in stasis, removed sleeper chems

### PR DESCRIPTION
**Pull request by JJRcop**
Revives #32535, even though @Iamgoofball might not be into it anymore. Let me know if you want your name off the changelog 😄 

:cl: JJRcop and Iamgoofball
add: Sleepers now place their occupants into stasis, which halts biological functions, like bleeding, disease, breathing, metabolism, et cetera.
del: They no longer hold chemicals, and no longer can inject chemicals.
add: This only affects station-based sleepers, others like syndicate, and clockwork sleepers, retain their chems.
/:cl:

So normal sleepers had their chems removed and I made them place occupants into stasis instead. Short of somehow getting attacked while inside the pod, stasis stops everything, including movement. This means you can place a critical patient in a sleeper to pause their ailments while you get treatment prepared. You can still resist out of a sleeper to get out, and while in stasis, one can still speak.

Other sleepers like clockwork and syndie retain chems, but the station ones don't.

I think this would be a good change to the game, because doctors will rely on other departments like chemistry and cargo to actually heal people, while the sleepers can be used to just throw someone in when you're overwhelmed or don't have the resources to heal them yet.

### CODE REVIEW

**Comment by WJohn** [comment]
Do syndie/clockwork sleepers also put you in stasis IN ADDITION to allowing chem injection or are they exactly the same as they are now?

**Comment by Grazyn** [comment]
what about mining sleepers? They kinda need the meds

**Comment by CPTANT** [comment]
So why would I ever put someone in a sleeper instead of a cryo tube?

**Comment by Bombpants** [comment]
How about they only have epinephrine at the base tier and require a large amount of upgrades just to get basic chems?

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -9,23 +9,27 @@
...
+		if(..()) //not dead
+			handle_blood()
 
-	if(stat != DEAD)
-		handle_liver()
+		if(stat != DEAD)
+			handle_liver()
+
+		//Updates the number of stored chemicals for powers
+		handle_changeling()
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/life.dm]: Worst to take a ling out of the game imo (for reference, this is under the !IsInStasis() check)

File: code/modules/mob/status_procs.dm [hunk 2]
```diff
@@ -118,6 +118,15 @@
 	..()
 	update_canmove()
 
+/////////////////////////////////// STASIS ///////////////////////////////////
+
+/mob/proc/IsInStasis()
+/mob/living/IsInStasis()
```
Comment (Cyberboss) [inline code/modules/mob/status_procs.dm]: return FALSE for the first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by pantherpusher** [comment]
Does the stasis fail when the sleeper doesn't have power?

File: code/modules/mob/status_procs.dm (hunk 2 above)
Comment (JJRcop) [reply code/modules/mob/status_procs.dm]: Is `. =  FALSE` okay? (byond's `return` has weird overhead)

File: code/modules/mob/living/carbon/life.dm (hunk 1 above)
Comment (JJRcop) [reply code/modules/mob/living/carbon/life.dm]: Will revert

**Comment by JJRcop** [comment]
@WJohn At the moment, they put you in stasis in addition to chems. I was mulling over this for 15~ minutes and decided to leave it like that.

My line of thinking is that you can put someone in the sleeper, put chems in, then simply open the sleeper to allow the chems you injected to metabolize. I was also considering adding a stasis button to the menu that would let you turn it off while they are still inside, so I could do that.

**Comment by JJRcop** [comment]
Maybe I could make the sleeper computer board eaaier to get, with this change.

**Comment by Grazyn** [comment]
>Do you mean survival pod sleepers? I can put some other form of meds in the pod, like a NanoMed, I don't want to add chems to them, though.

that would work but it's gonna be a special nanomed because they require medbay access

**Comment by kevinz000** [comment]
>not putting the fact that you're removing sleeper chemsi n pr title
0/10

**Comment by CPTANT** [comment]
So it IS literally an inferio cryo pod now.

**Comment by Qbopper** [comment]
sleepers putting people in stasis is a good change and thematically cool

removing sleeper chems is still making doctors entirely reliant on the whims of chemists at a time when it's a struggle to get people to even ask others for help, because they'd rather just break in and do it themselves

so unless a way to have doctors not use up the very limited medbay storage items/force the chemists to pay attention to medbay's requests is added at the same time it's likely not a great change

File: code/modules/mob/status_procs.dm (hunk 2 above)
Comment (JJRcop) [reply code/modules/mob/status_procs.dm]: resolved

**Comment by Iatots** [comment]
>people aren't doing their job
>let's buff object X to compensate for it

last 3 years of development basically

**Comment by PaznerIV** [comment]
are you not worried that people will rely on cyro or cloning more after this change? 
Because im pretty sure that what doctors/everyone else is just going to do instead of going to chem to heal.

**Comment by JJRcop** [comment]
I'm not worried about that, if players pivot over to that technique, that's fine. However, they won't have infinite heal chems anymore, which is the main idea behind this, while also keeping sleepers useful.

**Comment by Qbopper** [comment]
what i expect to happen as somebody who's played a fair bit of medbay

-sleepers will be used for the first few days because people didn't know about the change and then rarely ever used again
-since the easy heals from sleepers are gone doctors will start to rely on cryo for low effort heals
-an over-reliance on cryo means patients will be even more likely to just break into medbay storage for their meds instead of spending what feels like an hour unconscious
-a couple of chemist players will start making more meds knowing that medbay will need them, but doctors likely won't bother asking (either not thinking to or figuring chemistry won't bother, or whatever) and most chemists won't bother making meds

this is all just my predictions and there's a good chance I could be wrong but medbay is one of my favorite roles and beyond chemists making mannitol at round start or cures for disease outbreaks I can't remember the last time I saw chemists actually producing medicine

that being said I haven't been able to play as doctor in a little while, so for all I know chemists are extremely helpful by making a lot of medicines and are extremely cooperative with doctors now

**Comment by Jalleo** [comment]
@JJRcop can you make it so there is a small downside that causes a slight problem if the sleeper isn't a upgraded version (to keep a reqson for upgrading them)

**Comment by JJRcop** [comment]
@Jalleo I am trying to think of downsides, but I don't want to compromise on the stasis effect, so it should be something else. Do you have any ideas?

**Comment by ShizCalev** [comment]
Conflict.

**Comment by PKPenguin321** [comment]
Allow me to fill the sleeper with my own chems from chemistry and label them whatever I want and allow the sleeper to inject the patient with these custom chems

You can
>Insert homemade bicaridine and retain old functionality while still requiring interaction from chemistry and not making it free
>Label healing chems as "heals brute," "heals burn," etc
>Label poison as "heals brute," "heals burn," etc as a sick prank

**Comment by JJRcop** [comment]
@PKPenguin321 Brilliant! May take some time as my priorities are different than they were before.

**Comment by Qbopper** [comment]
i thought that idea was proposed last time and got shot down for (???) reasons

**Comment by davethwave** [comment]
a mild stamina damage that effects the person for a minute or 2 perhaps without the proper upgrades?

**Comment by JJRcop** [comment]
Don't know when I will work on this, will reopen when I do and finish.

**Pull request closed without merging**